e2e/helpers/http.ts

```diff
@@ -1,141 +1,107 @@
-}
-
-function generateDigitalSignature(
-  data: string,
-  base64EncodedSecretKey: string,
-) {
-  const hash = createHmac(
-    'sha1',
-    Buffer.from(base64EncodedSecretKey, BASE64_ENCODING),
-  )
-    .update(data)
-    .digest(BASE64_ENCODING)
-  return hash
-}
-
-function getDigitalSignature(
-  uri: string,
-  username: string,
-  base64EncodedSecretKey: string,
-) {
-  const timestamp = new Date().toISOString()
-  const signature = generateDigitalSignature(
-    `${username}${uri}${timestamp}`,
-    base64EncodedSecretKey,
-  )
-  return {
-    userId: username,
-    signatureTimestamp: timestamp,
-    signature: signature,
-  }
-}
-
-export function getUserIdFromJwt(token: string) {
-  const payload = JSON.parse(
-    Buffer.from(token.split('.')[1], BASE64_ENCODING).toString(),
-  )
-  return payload.sub
+export enum BackendDestinationEnum {
+  REPO_ENDPOINT,
+  PORTAL_ENDPOINT,
 }
 
-function updateHeaders(
-  headers: { [key: string]: string },
-  uri: string,
-  accessToken?: string,
-  userName?: string,
-  apiKey?: string,
-) {
-  return {
-    ...headers,
-    ...(accessToken && {
-      'Access-Control-Request-Headers': 'authorization',
-      Authorization: `Bearer ${accessToken}`,
-    }),
-    ...(apiKey && userName && getDigitalSignature(uri, userName, apiKey)),
-  }
+export async function waitForSrcEndpointConfig(page: Page) {
+  // window only available after page has initially loaded
+  await navigateToHomepageIfPageHasNotBeenLoaded(page)
+  // ensure that endpoint config is set,
+  // ...so API calls point to the correct stack
+  await expect(async () => {
+    const response = await page.evaluate('window.SRC.OVERRIDE_ENDPOINT_CONFIG')
+    expect(response).not.toBeUndefined()
+  }).toPass()
 }
 
 export async function doPost<T>(
-  endpoint: string,
-  uri: string,
-  requestContent: string,
-  accessToken?: string,
-  userName?: string,
-  apiKey?: string,
+  page: Page,
+  url: string,
+  requestJsonObject: unknown,
+  accessToken: string | undefined,
+  endpoint: BackendDestinationEnum,
+  additionalOptions: RequestInit = {},
 ) {
-  const url: RequestInfo = `${endpoint}${uri}`
-  const options: RequestInit = {
-    body: requestContent,
-    headers: updateHeaders(
-      DEFAULT_POST_HEADERS,
-      uri,
+  await waitForSrcEndpointConfig(page)
+  const response = await page.evaluate(
+    async ({
+      url,
+      requestJsonObject,
       accessToken,
-      userName,
-      apiKey,
-    ),
-    method: 'POST',
-    mode: 'cors',
-  }
-  return await fetchWithExponentialTimeout(url, options)
+      endpoint,
+      additionalOptions,
+    }) => {
+      // @ts-expect-error: Cannot find name 'SRC'
+      const srcEndpoint = await SRC.SynapseEnums.BackendDestinationEnum[
+        endpoint
+      ]
+      // @ts-expect-error: Cannot find name 'SRC'
+      return await SRC.HttpClient.doPost(
+        url,
+        requestJsonObject,
+        accessToken,
+        srcEndpoint,
+        additionalOptions,
+      )
```

Comment:
Maintain a `doPost` method that evaluates the page to call the `SRC.HttpClient.doPost` method to reduce code duplication in API calls. Plan to remove this method (and `doGet` and `doDelete`) once SynapseClient is its own package that can be used directly instead